install_files/securedrop-app-code/DEBIAN/preinst

```diff
@@ -13,12 +13,46 @@ set -e
 # for details, see http://www.debian.org/doc/debian-policy/ or
 # the debian-policy package
 
+
+function permanently_disable_swap() {
+    # Swap usage is prohibited in the context of SecureDrop, due to risk of
+    # forensic discovery recovering plaintext submissions that were written
+    # to disk prior to encrypting. If a swapfile is active, disable it,
+    # and disable all swap entries from /etc/fstab, to prevent mounting
+    # after reboots.
+
+    # Check if swapfile is currently enabled.
+    swap_result=$(swapon --summary)
+
+    # Search for path format in output.
+    swap_partition=$(grep '/' <<< "${swap_result}" | perl -ane 'print $F[0]')
+
+    # If a filepath for swap was found, disable it.
+    if [[ -n "${swap_partition}" ]]; then
+        echo "Disabling swap..."
+        # Disable all active swap.
+        swapoff --all
+        # Securely erase swap partition.
+        shred "${swap_partition}"
```

Comment:
(`swap partition size` * `3`) / `sequential write speed`. 5 minutes max with low-end hardware (80MB/s HDD) and a big swap partition (8GB). Except that could be wrong if `shred` uses a blocking randomness source. I'm going to go with we probably don't need a warning and for most people using SSDs it will be about 30s. We can test on our NUC.